downstairs/src/region.rs

```diff
@@ -799,36 +812,22 @@ impl Region {
         /*
          * Before anything, validate hashes
          */
-        self.validate_hashes(writes)?;
-
-        /*
-         * Batch writes so they can all be sent to the appropriate extent
-         * together.
-         */
-        let mut batched_writes: HashMap<usize, Vec<crucible_protocol::Write>> =
-            HashMap::new();
-
-        for write in writes {
-            let extent_vec =
-                batched_writes.entry(write.eid as usize).or_default();
-            extent_vec.push(write.clone());
-        }
+        self.validate_hashes(&writes)?;
 
         if only_write_unwritten {
             cdt::os__writeunwritten__start!(|| job_id.0);
         } else {
             cdt::os__write__start!(|| job_id.0);
         }
-        for eid in batched_writes.keys() {
-            let extent = self.get_opened_extent_mut(*eid);
-            let writes = batched_writes.get(eid).unwrap();
+        for req in writes {
+            let extent = self.get_opened_extent_mut(req.extent as usize);
             extent
-                .write(job_id, &writes[..], only_write_unwritten)
+                .write(job_id, req.write, only_write_unwritten)
                 .await?;
-        }
 
-        // Mark any extents we sent a write-command to as potentially dirty
-        self.dirty_extents.extend(batched_writes.keys());
+            // Mark any extents we sent a write-command to as potentially dirty
```

Comment:
Is this in the correct place?  (It was here before you got here) but I'm wonding
if the mark as dirty should happen before we do the actual write?

Reply:
Interesting – I don't know whether a failed write should qualify an extent for sync-to-disk, but it's probably safer.  I moved this above the write attempt in b9b087e1f731

Reply:
I don't think this is right, though the problem might have existed before this PR too - if all writes are skipped, then `set_dirty` isn't called. `dirty_extents`' comment says

> Should be true if the dirty flag in the extent's metadata is set.

which might drift in that case.